apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -0,0 +1,337 @@
+                await client.GetAsync($"/Main/Critical/{testId}");
+            }
+
+            // NoBufferLoggerTestApplication does not support debug or info logs.
+            var results = GetEntries(startTime, testId, 3);
+            Assert.Equal(3, results.Count());
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Error));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+
+        [Fact]
+        public async Task Logging_SizedBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<SizedBufferErrorLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                for (int i = 0; i < 250; i++)
+                {
+                    await client.GetAsync($"/Main/Debug/{testId}");
+                    await client.GetAsync($"/Main/Info/{testId}");
+                    await client.GetAsync($"/Main/Warning/{testId}");
+                    await client.GetAsync($"/Main/Error/{testId}");
+                    await client.GetAsync($"/Main/Critical/{testId}");
+                    await client.GetAsync($"/Main/Exception/{testId}");
+                }
+            }
+
+            // Just check that a large portion of logs entires were pushed.  Not all
+            // will be pushed as some may be in the buffer.
+            var results = GetEntries(startTime, testId, 500);
+            Assert.True(results.Count() >= 500);
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Debug));
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Info));
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Error));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+
+
+        [Fact]
+        public async Task Logging_TimedBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<TimedBufferWarningLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+
+                await client.GetAsync($"/Main/Warning/{testId}");
+                await client.GetAsync($"/Main/Error/{testId}");
+
+                var noResults = GetEntries(startTime, testId, 0);
+                Assert.Empty(noResults);
+                Thread.Sleep(TimeSpan.FromSeconds(35));
```

Comment:
This still seems like a while. Is it possible to use a timed buffer of 5 seconds and then sleep for 7? Or does that make things too flaky?